include/rocksdb/configurable.h

```diff
@@ -356,4 +356,8 @@ class Configurable {
   // Configurable option via
   std::vector<RegisteredOptions> options_;
 };
+
+extern void RegisterConfigurableOptions(
+    Configurable& configurable, const std::string& name, void* opt_ptr,
+    const std::unordered_map<std::string, OptionTypeInfo>* opt_map);
```

Comment:
I am not sure what the purpose of exposing this method is.  What are you hoping to accomplish?

I do not see where this method is used anywhere in this check-in.  Perhaps this should be moved into a separate PR?

Reply:
The purpose of this PR is to enable 3rd party implementations of ```TieredCache```. The URI will contain the type, as well as any options required to instantiate the object. For example, you could have "cachelib://path_to_cache_file/size=1G;buffers=10M;io_threads=16". We currently allow 3rd party implementations of Env etc. to register a factory function with ```ObjectLibrary``` for the URI pattern, but I couldn't find any support for passing option name-value pairs.

Let me know if there's a better way. I'm open to any suggestions.

Reply:
@mrambacher do your future PRs on Configurable be able to handle the configuration from the URI string?

Reply:
@zhichao-cao @anand1976 Yes, it would be a straightforward exercise to make TieredCache into a Customizable object that can be configured from an input string.  The string would look slightly different than what may be proposed here (id=TieredCacheClass; prop1=value1; etc), but it should be easy to implement.